plugins/woocommerce/tests/e2e-pw/tests/merchant/product-delete.spec.js

```diff
@@ -0,0 +1,121 @@
+const {test: baseTest, expect} = require('@playwright/test');
+const wcApi = require('@woocommerce/woocommerce-rest-api').default;
+
+baseTest.describe('Products > Delete Product', () => {
+	baseTest.use({storageState: process.env.ADMINSTATE});
+
+	const apiFixture = baseTest.extend({
+		api: async ({baseURL}, use) => {
+			const api = new wcApi({
+				url: baseURL,
+				consumerKey: process.env.CONSUMER_KEY,
+				consumerSecret: process.env.CONSUMER_SECRET,
+				version: 'wc/v3',
+				axiosConfig: {
+					// allow 404s, so we can check if the product was deleted without try/catch
+					validateStatus: function (status) {
+						return status >= 200 && status < 300 || status === 404;
+					},
+				},
+			});
+
+			await use(api);
+		},
+	});
+
+	const test = apiFixture.extend({
+		productId: async ({baseURL, api}, use) => {
+			let productId;
+
+			await api
+				.post('products', {
+					name: 'Product to delete',
+					type: 'simple',
+					regular_price: '12.99',
+				})
+				.then((response) => {
+					productId = response.data.id;
+				});
+
+			await use(productId);
+
+			// permanently delete the product if it still exists
+			const r = await api.get(`products/${productId}`);
+			if (r.status !== 404) {
+				await api.delete(`products/${productId}`, {
+					force: true,
+				});
+			}
+		},
+	});
+
+	test('can delete a product from edit view', async ({page, productId, api}) => {
+		await test.step('Navigate to product edit page', async () => {
+			await page.goto(`wp-admin/post.php?post=${productId}&action=edit`);
+		});
+
+		await test.step('Move product to trash', async () => {
+			await page.locator('#delete-action > a').click();
+		});
+
+		await test.step('Verify product was trashed', async () => {
+			await expect(
+				page.locator('#message').last()
+			).toContainText('1 product moved to the Trash.');
+
+			const response = await api.get(`products/${productId}`);
+			await expect(response.data.status).toBe('trash');
+		});
+	});
+
+	test('can quick delete a product from product list', async ({page, productId, api}) => {
+		await test.step('Navigate to products list page', async () => {
+			await page.goto(`wp-admin/edit.php?post_type=product`);
+		});
+
+		await test.step('Move product to trash', async () => {
+			// mouse over the product row to display the quick actions
+			await page.locator(`#post-${productId}`).hover();
+
+			// move product to trash
+			await page.locator(`#post-${productId} .submitdelete`).click();
+		});
+
+		await test.step('Verify product was trashed', async () => {
+			await expect(
+				page.locator('#message').last()
+			).toContainText('1 product moved to the Trash.');
+
+			const response = await api.get(`products/${productId}`);
+			await expect(response.data.status).toBe('trash');
```

Comment:
I think `await` has no action on this type of expression. Can be removed.

Reply:
Removed with https://github.com/woocommerce/woocommerce/pull/43518/commits/791700dd19a7693fa22a93e6e2a22d6ed970b8c8